test/test_storage.py

```diff
@@ -33,27 +33,55 @@
 import ops.storage
 
 
+@pytest.fixture
+def fake_script(request: pytest.FixtureRequest):
+    return FakeScript(request)
+
+
 class StoragePermutations(abc.ABC):
 
     assertEqual = unittest.TestCase.assertEqual  # noqa
     assertRaises = unittest.TestCase.assertRaises  # noqa
 
-    def create_framework(self) -> ops.Framework:
+    def create_framework(
+        self,
+        request: pytest.FixtureRequest,
+        fake_script: FakeScript,
+    ) -> ops.Framework:
         """Create a Framework that we can use to test the backend storage."""
-        return ops.Framework(self.create_storage(), None, None, None)  # type: ignore
+        return ops.Framework(
+            self.create_storage(
+                request,
+                fake_script
+            ),
+            None,  # type: ignore
+            None,  # type: ignore
+            None)  # type: ignore
```

Comment:
I think the formatting could be significantly improved by doing this on two lines:

```suggestion
        storage = self.create_storage(request, fake_script)
        return ops.Framework(storage, None, None, None)  # type: ignore
```